adafruit_tmp007.py

```diff
@@ -0,0 +1,197 @@
+
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_TMP007.git"
+
+
+# Default device I2C address.
+_TMP007_I2CADDR = const(0x40)
+
+# Register addresses.
+_TMP007_CONFIG = const(0x02)
+_TMP007_DEVID = const(0x1F)
+_TMP007_VOBJ = const(0x0)
+_TMP007_TAMB = const(0x01)
+_TMP007_TOBJ = const(0x03)
+
+# Config register values.
+_TMP007_CFG_RESET = const(0x8000)
+_TMP007_CFG_MODEON = const(0x7000)
+CFG_1SAMPLE = const(0x0000)
+CFG_2SAMPLE = const(0x0200)
+CFG_4SAMPLE = const(0x0400)
+CFG_8SAMPLE = const(0x0600)
+CFG_16SAMPLE = const(0x0800)
+_TMP007_CFG_DRDYEN = const(0x0100)
+_TMP007_CFG_DRDY = const(0x0080)
+
+
+class TMP007:
+    """Class to represent an Adafruit TMP007 non-contact temperature measurement
+    board.
+    """
+
+    # Class-level buffer for reading and writing data with the sensor.
+    # This reduces memory allocations but means the code is not re-entrant or
+    # thread safe!
+    _BUFFER = bytearray(4)
+
+
+
+    def __init__(self, i2c, address=_TMP007_I2CADDR):
+        """Initialize TMP007 device on the specified I2C address and bus number.
+        Address defaults to 0x40 and bus number defaults to the appropriate bus
+        for the hardware.
+        """
+        self._device = I2CDevice(i2c, address)
+
+    def begin(self, samplerate=CFG_16SAMPLE):
+        """Start taking temperature measurements.  Samplerate can be one of
+        TMP007_CFG_1SAMPLE, TMP007_CFG_2SAMPLE, TMP007_CFG_4SAMPLE,
+        TMP007_CFG_8SAMPLE, or TMP007_CFG_16SAMPLE.  The default is 16 samples
+        for the highest resolution.  Returns True if the device is intialized,
+        False otherwise.
+        """
+        self._write_u16(_TMP007_CONFIG, _TMP007_CFG_RESET)
+        time.sleep(.5)
+        if samplerate not in (CFG_1SAMPLE, CFG_2SAMPLE, CFG_4SAMPLE, CFG_8SAMPLE,
+                              CFG_16SAMPLE):
+            raise ValueError('Unexpected samplerate value! Must be one of: ' \
+                'CFG_1SAMPLE, CFG_2SAMPLE, CFG_4SAMPLE, CFG_8SAMPLE, or CFG_16SAMPLE')
+        # Set configuration register to turn on chip, enable data ready output,
+        # and start sampling at the specified rate.
+        config = _TMP007_CFG_MODEON | _TMP007_CFG_DRDYEN | samplerate
+        self._write_u16(_TMP007_CONFIG, config)
+        # Check device ID match expected value.
+        return  self.read_register(_TMP007_DEVID) == 0x0078
+
+
+
+    def sleep(self):
+        """Put TMP007 into low power sleep mode.  No measurement data will be
+        updated while in sleep mode.
+        """
+        control = self._read_u16(_TMP007_CONFIG)
+        control &= ~(_TMP007_CFG_MODEON)
+        self._write_u16(_TMP007_CONFIG, control)
+
+    def wake(self):
+        """Wake up TMP007 from low power sleep mode."""
+        control = self._read_u16(_TMP007_CONFIG)
+        control |= _TMP007_CFG_MODEON
+        self._write_u16(_TMP007_CONFIG, control)
+
+    def read_raw_voltage(self):
```

Comment:
Please use properties for sensor values. I'd expect `temperature` for the object and `sensor_temperature` for the die temp. voltage can be private only.

Reply:
made properties -- the voltage is not actually used internally. I think it is to be available for detailed sensor calibration by the user.